[10.0] Port account_invoice_rounding

grindtildeath opened the pull request.

grindtildeath committed:
Port account_invoice_rounding to 10.0

leemannd commented on account_invoice_rounding/models/account.py:
```diff
@@ -18,23 +18,23 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Short header

leemannd commented on account_invoice_rounding/models/company.py:
```diff
@@ -18,7 +18,7 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
```
Short headers

leemannd commented on account_invoice_rounding/tests/test_invoice_rounding.py:
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+import openerp.tests.common as test_common
```
s/openerp/odoo/

leemannd reviewed (changes requested):
Thanks for the port. Only nitpicking

leemannd committed:
Fix swedish_rounding singleton error

Commit:
[FIX] Fix compute_amount super and installation on readme.rst

hurrinico commented:
i've create [this](https://github.com/grindtildeath/account-invoicing/pull/3) in order to fix a bug using globally swedish round and fix the readme.rst.
To reproduce the bug:
Go to runbot.
set swedish round global.
create invoice add a product and set unit price 2,99 with tax 10%
try to add another line in order to trigger the onchange, you can see tax set as 0,31 and the total as 3,30
if you save tax goes to 0,01 and total to 3,00.
This PR fix it

grindtildeath commented:
@hurrinico Thanks for your PR.
I could reproduce the bug on runbot but not on my local instance because I used an `account.tax` having `price_included = True`.
Then I tried setting this flag to False, and could reproduce the bug you reported without your fix.
However, after pulling your branch the behaviour is still the same using `account.tax` without price_included :confused: 
As it's friday 5.30pm passed, I won't look more into it now, and will try to spend some time on it next week. :sweat_smile:

grindtildeath committed:
Fix rounding error due to recursive call to write

Use invoice2 var cache to assert tax amount because self is not right :/

grindtildeath commented:
@hurrinico @leemannd @guewen 
PR fixed, can you please review ?

hurrinico commented:
there's a problem, travis is failing cause of api.one, but we need it in order to fix a bug

grindtildeath commented:
@hurrinico Actually flake8 check is failing because of this line https://github.com/OCA/account-invoicing/blob/10.0/account_payment_term_extension/models/account_payment_term.py#L70
So it has nothing to do with this PR.

hurrinico commented:
@grindtildeath maybe a rebase to OCA/10 will fix it.
I suggest also to squash commits by author, in order to have 2 commits here one mine one yours

grindtildeath commented:
@hurrinico We will need this https://github.com/OCA/account-invoicing/pull/324 to be merged before a rebase.
I will wait on it to squash commits at the same time.

hurrinico commented:
@grindtildeath great :+1: after the fix

tafaRU commented:
@grindtildeath, you could rebase on top considering that https://github.com/OCA/account-invoicing/pull/324 now is merged.
Thank you in advance.

hurrinico commented:
great work @grindtildeath thank  you for the fix :+1:

grindtildeath committed:
Fix onchange breaking init of fields in accounting settings

grindtildeath commented:
@hurrinico @tafaRU Could you please review so we can get this merged ?

eLBati committed:
FIX

account_invoice_rounding/models/account.py", line 102, in _compute_swedish_rounding
    if round_method[:7] != 'swedish':
TypeError: 'bool' object has no attribute '__getitem__'

when no rounding method is set

eLBati commented:
@grindtildeath please see https://github.com/grindtildeath/account-invoicing/pull/4

grindtildeath committed:
Merge pull request #4 from eLBati/10.0_mig_account_invoice_rounding-lb

FIX

grindtildeath commented:
@pedrobaeza This PR is hanging still for a while and the module is already in use on a few installations, shouldn't we merge it ?

pedrobaeza commented:
We need another reviewer for this one. Can you ask any of your colleagues to review it, or maybe you review another PR and ask in exchange that he/she reviews yours?

pedrobaeza commented:
Sorry, I didn't see the first one. We have 2 approvals, so I can merge.
